sdk/core/core/internal/az_url_internal.h

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef _az_URL_INTERNAL_H
+#define _az_URL_INTERNAL_H
+
+#include <az_config.h>
+#include <az_result.h>
+#include <az_span.h>
+
+#include <stdint.h>
+
+#include <_az_cfg_prefix.h>
+
+/**
+ * @brief Copies character from the \p source #az_span to the \p destination #az_span by
+ * URL-encoding the \p source span characters.
+ *
+ * @param[in] destination The #az_span whose bytes will receive the URL-encoded source.
+ * @param[in] source The #az_span containing the non-URL-encoded bytes.
+ * @param[out] out_length A pointer to a #uint32_t that is going to be assigned the length
```

Comment:
It isn't a uint32_t.
```suggestion
 * @param[out] out_length A pointer to an int32_t that is going to be assigned the length
```